client/lib/tree-select/index.js

```diff
@@ -47,8 +54,19 @@ export default function treeSelect( getDependents, selector ) {
 		leafCache.set( key, value );
 		return value;
 	};
+
+	cachedSelector.clearCache = () => {
+		cache = new WeakMap();
+	};
```

Comment:
this needs a comment explaining that `WeakMap` has no `.clear()` _or_ we should use an object that is like a `WeakMap` that _does_ have `.clear()` - someone's bound to change this otherwise when they don't realize that

Reply:
Added a comment.